tests/test_remove_repo.py

```diff
@@ -0,0 +1,358 @@
+        "dummy",
+        "--repository",
+        "namespace/image",
+        "--namespace",
+        "internal-namespace",
+        "--quay-user",
+        "some-user",
+        "--quay-password",
+        "some-password",
+        "--quay-api-token",
+        "some-token",
+        "--pyxis-server",
+        "pyxis-url.com",
+        "--pyxis-krb-principal",
+        "some-principal",
+    ]
+    mock_delete_repo = mock.MagicMock()
+    mock_quay_api_client.return_value.delete_repository = mock_delete_repo
+    mock_set_quay_api_client = mock.MagicMock()
+    mock_remove_repository_signatures = mock.MagicMock()
+    mock_signature_remover.return_value.set_quay_api_client = mock_set_quay_api_client
+    mock_signature_remover.return_value.remove_repository_signatures = (
+        mock_remove_repository_signatures
+    )
+
+    remove_repo.remove_repository_main(args)
+
+    mock_quay_api_client.assert_called_once_with("some-token")
+    mock_signature_remover.assert_called_once_with(
+        quay_user="some-user", quay_password="some-password"
+    )
+    mock_set_quay_api_client.assert_called_once_with(mock_quay_api_client.return_value)
+    mock_remove_repository_signatures.assert_called_once_with(
+        "namespace/image", "internal-namespace", "pyxis-url.com", "some-principal", None
+    )
+    mock_delete_repo.assert_called_once_with("namespace/image")
+    mock_send_umb_message.assert_not_called()
+
+
+@mock.patch("pubtools._quay.remove_repo.SignatureRemover")
+@mock.patch("pubtools._quay.remove_repo.send_umb_message")
+@mock.patch("pubtools._quay.remove_repo.QuayApiClient")
+def test_send_umb_message(mock_quay_api_client, mock_send_umb_message, mock_signature_remover):
+    args = [
+        "dummy",
+        "--repository",
+        "namespace/image",
+        "--namespace",
+        "internal-namespace",
+        "--quay-user",
+        "some-user",
+        "--quay-password",
+        "some-password",
+        "--quay-api-token",
+        "some-token",
+        "--pyxis-server",
+        "pyxis-url.com",
+        "--pyxis-krb-principal",
+        "some-principal",
+        "--send-umb-msg",
+        "--umb-url",
+        "amqps://url:5671",
+        "--umb-cert",
+        "/path/to/file.crt",
+        "--umb-client-key",
+        "/path/to/umb.key",
+        "--umb-ca-cert",
+        "/path/to/ca_cert.crt",
+        "--umb-topic",
+        "VirtualTopic.eng.pub.remove_repo_new",
+    ]
+    mock_delete_repo = mock.MagicMock()
+    mock_quay_api_client.return_value.delete_repository = mock_delete_repo
+    mock_set_quay_api_client = mock.MagicMock()
+    mock_remove_repository_signatures = mock.MagicMock()
+    mock_signature_remover.return_value.set_quay_api_client = mock_set_quay_api_client
+    mock_signature_remover.return_value.remove_repository_signatures = (
+        mock_remove_repository_signatures
+    )
+    remove_repo.remove_repository_main(args)
+
+    mock_quay_api_client.assert_called_once_with("some-token")
```

Comment:
This is already tested in previous test, so you don't need to test it here again

Reply:
removed